docs/orchestrating-elastic-stack-applications/enterprise-search.asciidoc

```diff
@@ -0,0 +1,242 @@
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: quickstart
+  config:
+    ent_search.external_url: https://my-custom-domain:3002
+  configRef:
+  - secretName: smtp-credentials
+---
+kind: Secret
+apiVersion: v1
+metadata:
+  name: smtp-credentials
+stringData:
+  enterprise-search.yml: |-
+    email.account.enabled: true
+    email.account.smtp.auth: plain
+    email.account.smtp.starttls.enable: false
+    email.account.smtp.host: 127.0.0.1
+    email.account.smtp.port: 25
+    email.account.smtp.user: myuser
+    email.account.smtp.password: mypassword
+    email.account.email_defaults.from: my@email.com
+----
+
+ECK merges the content of `config` and `configRef` into a single internal Secret. In case of duplicate settings, the last referenced Secret takes precedence.
+
+[id="{p}-enterprise-search-custom-pod-template"]
+=== Customize the Pod template
+
+Enterprise Search Pods specification can be overridden through the `podTemplate` element.
+
+This example overrides the default 4Gi deployment to use 8Gi instead, and makes the deployment highly-available with 3 Pods:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: enterprisesearch.k8s.elastic.co/v1beta1
+kind: EnterpriseSearch
+metadata:
+  name: enterprise-search-quickstart
+spec:
+  version: {version}
+  count: 3
+  elasticsearchRef:
+    name: quickstart
+  podTemplate:
+    spec:
+      containers:
+      - name: enterprise-search
+        resources:
+          requests:
+            cpu: 3
+            memory: 8Gi
+          limits:
+            memory: 8Gi
+        env:
+        - name: JAVA_OPTS
+          value: -Xms7500m -Xmx7500m
+----
+
+[id="{p}-enterprise-search-expose"]
+=== Expose Enterprise Search
+
+By default ECK manages self-signed TLS certificates to secure the connection to Enterprise Search.
+It also restricts the Kubernetes service to `ClusterIP` type that cannot be accessed publicly.
+
+See link:../k8s-accessing-elastic-services.html[how to access Elastic Stack services] to customize TLS settings and expose the service.
+
+NOTE: When exposed outside the scope of `localhost`, make sure to set `ent_search.external_url` accordingly in Enterprise Search configuration.
+
+[id="{p}-enterprise-search-connect-non-eck-es"]
+=== Connect to an external Elasticsearch cluster
+
+The `elasticsearchRef` element allows ECK to automatically configure Enterprise Search to establish a secured connection to a managed Elasticsearch cluster.
+
+Enterprise Search can also be manually configured to access any available Elasticsearch cluster:
+
+[source,yaml,subs="attributes,+macros"]
+----
+cat $$<<$$EOF | kubectl apply -f -
+apiVersion: enterprisesearch.k8s.elastic.co/v1beta1
+kind: EnterpriseSearch
+metadata:
+  name: enterprise-search-quickstart
+spec:
+  version: {version}
+  count: 1
+  config:
+    elasticsearch.host: https://elasticsearch-url:9200
+    elasticsearch.username: elastic
+    elasticsearch.password: my-password
```

Comment:
I don't think we want to encourage this behavior, correct?

Reply:
do you mean cleartext credentials, or association to an external cluster?

For the first one, I should definitely make this a `configRef` secret.

Reply:
Yes the cleartext credentials is what I meant, sorry for not being clearer the first time around